server/src/main/java/org/elasticsearch/cluster/service/TransportVersionsFixupListener.java

```diff
@@ -99,7 +101,9 @@ public ClusterState execute(BatchExecutionContext<NodeTransportVersionTask> cont
             for (var c : context.taskContexts()) {
                 for (var e : c.getTask().results().entrySet()) {
                     // this node's transport version might have been updated already/node has gone away
-                    TransportVersion recordedTv = builder.transportVersions().get(e.getKey());
+                    TransportVersion recordedTv = Optional.ofNullable(builder.compatibilityVersions().get(e.getKey()))
```

Comment:
Could use `Map.getOrDefault(key, new CompatVersion(null))` here to simplify it a bit

Reply:
I think that'll probably end up more complicated once we start to add new versions. At least we'll have to keep on adding `, null` to the constructor call.

Reply:
I agree with David, but it gets a little more readable if I make a variable for `builder.compatibilityVersions()`, so I'll do that.